File: 双指针/88/88.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
 var merge = function(nums1, m, nums2, n) {
  let p1 = m + n - 1;
  let pm = m - 1;
  let pn = n - 1;
  while(pm >= 0 && pn >= 0) {
      const vm = nums1[pm];
      const vn = nums2[pn];
      if (vn > vm) {
          nums1[p1] = vn;
          pn --;
      } else {
          nums1[p1] = vm;
          pm --;
      }
      p1 --;
  }
  if (pn >= 0) {
      for (;pn >=0 ; pn --) {
          nums1[p1] = nums2[pn];
          p1 --;
      }
  }
};

File: 双指针/88/88.c
void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n){
    int p1 = nums1Size - 1;
    int pm = m - 1;
    int pn = n - 1;
    while(pm >= 0 && pn >= 0) {
        int vm = nums1[pm];
        int vn = nums2[pn];
        if (vm > vn) {
            nums1[p1] = vm;
            pm --;
        } else {
            nums1[p1] = vn;
            pn --;
        }
        p1 --;
    }
    if (pn >= 0) {
        for (int i = 0; i <= pn ; i ++) {
            nums1[i] = nums2[i];
        }
    }
}